addons/mail/static/src/messaging/component/composer_text_input/composer_text_input.js

```diff
@@ -161,14 +339,114 @@ class ComposerTextInput extends Component {
+                    ev.preventDefault();
+                }
+                break;
+            // ENTER: submit the message only if the dropdown mention proposition is not displayed
             case 'Enter':
-                this._onKeydownTextareaEnter(ev);
+                if (this.state.isMentionsOpen) {
+                    ev.preventDefault();
+                } else {
+                    this._onKeydownTextareaEnter(ev);
+                }
                 break;
+        }
+    }
+
+    /**
+     * @private
+     * @param ev
+     */
+    _onKeyupTextarea(ev) {
+        let active = undefined;
+        let selectionIndex = undefined;
+        if (this.state.isMentionsOpen) {
+            const active_id = this.state.active_id;
+            active = this.state.mentionSuggestions[0].find(function (item) {
+                return item.id === active_id;
+            });
+            selectionIndex = this.state.mentionSuggestions[0].indexOf(active);
+        }
+        switch (ev.key) {
+            // ESCAPE: close mention propositions
             case 'Escape':
-                this._onKeydownTextareaEscape(ev);
+                if (this.state.isMentionsOpen) {
+                    this.state.isMentionsOpen = false;
+                    this.state.mentionSuggestions = [];
+                    this.state.active_id = 0;
+                    ev.stopPropagation();
+                } else {
+                    this._onKeyupTextareaEscape(ev);
+                }
                 break;
-            default:
+            // ENTER, HOME, END, UP, DOWN, PAGE UP, PAGE DOWN, TAB: check if navigation in mention propositions
+            case 'Enter':
+                if (this.state.isMentionsOpen) {
+                    this._insertMentionSuggestion(active);
+                }
                 break;
+            case 'ArrowUp':
+            case 'PageUp':
+                if (this.state.isMentionsOpen) {
+                    if (selectionIndex !== 0) {
+                        active = this.state.mentionSuggestions[0][selectionIndex - 1];
+                    } else {
+                        active = this.state.mentionSuggestions[0][this.state.mentionSuggestions[0].length - 1];
+                    }
+                    this.state.active_id = active.id;
+                }
+                break;
+            case 'ArrowDown':
+            case 'PageDown':
+                if (this.state.isMentionsOpen) {
+                    if (selectionIndex !== this.state.mentionSuggestions[0].length - 1) {
+                        active = this.state.mentionSuggestions[0][selectionIndex + 1];
+                    } else {
+                        active = this.state.mentionSuggestions[0][0];
+                    }
+                    this.state.active_id = active.id;
+                }
+                break;
+            case 'Home':
+                if (this.state.isMentionsOpen) {
+                    active = this.state.mentionSuggestions[0][0];
```

Comment:
not easier to read what 1st item of 1st item of array means.
Certainly need a better data structure here.